Sources/UnidirectionalBinding.swift

```diff
@@ -304,14 +305,17 @@ public final class BindingTarget<Value>: BindingTargetProtocol {
 	///   - setter: The action to receive values.
 	public convenience init(on queue: DispatchQueue, lifetime: Lifetime, setter: @escaping (Value) -> Void) {
 		let queueId = ObjectIdentifier(queue)
+		let usesMainQueue = queue == DispatchQueue.main
 
 		/// Ensures the queue has been setup property.
 		if nil == queue.getSpecific(key: specificKey) {
 			queue.setSpecific(key: specificKey, value: queueId)
 		}
 
 		let setter: (Value) -> Void = { value in
-			if queueId == DispatchQueue.getSpecific(key: specificKey) {
+			if usesMainQueue && Thread.isMainThread {
```

Comment:
We shouldn't be using the main thread: https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2912.

It seems like this API shouldn't be using queues at all—it should use schedulers. I don't know if that will have any affect on the issue at hand.